geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala

```diff
@@ -140,17 +140,24 @@ abstract class GeoMesaDataStore[DS <: GeoMesaDataStore[DS, F, W], F <: WrappedFe
           // do this before anything else so that any modifications will be in place
           GeoMesaSchemaValidator.validate(sft)
 
-          // write out the metadata to the catalog table
-          writeMetadata(sft)
-
-          // reload the sft so that we have any default metadata,
-          // then copy over any additional keys that were in the original sft
-          val reloadedSft = getSchema(sft.getTypeName)
-          (sft.getUserData.keySet -- reloadedSft.getUserData.keySet)
+          try {
+            // write out the metadata to the catalog table
+            writeMetadata(sft)
+
+            // reload the sft so that we have any default metadata,
+            // then copy over any additional keys that were in the original sft
+            val reloadedSft = getSchema(sft.getTypeName)
+            (sft.getUserData.keySet -- reloadedSft.getUserData.keySet)
               .foreach(k => reloadedSft.getUserData.put(k, sft.getUserData.get(k)))
 
-          // create the tables in accumulo
-          manager.indices(reloadedSft, IndexMode.Any).foreach(_.configure(reloadedSft, this))
+            // create the tables
+            manager.indices(reloadedSft, IndexMode.Any).foreach(_.configure(reloadedSft, this))
+          } catch {
+            case e: Exception =>
+              // If there was an error creating a schema, clean up.
+              metadata.delete(sft.getTypeName)
```

Comment:
i would wrap this in a try/catch as well - in the second catch block you can use `e.addSuppressed(e2)`